src/test/compile-fail-fulldeps/proc-macro/macros-in-extern.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// aux-build:test-macros.rs
+// ignore-stage1
+// ignore-wasm32
+
+#![feature(proc_macro)]
+
+extern crate test_macros;
+
+use test_macros::{nop_attr, no_output, emit_input};
+
+fn main() {
+    #[no_output]
```

Comment:
@petrochenkov possible regression introduced by #49124? This invocation is being feature-gated by `stmt_expr_attributes` but I don't think it should be.

Reply:
Hmm, I thought the idea was to additionally gate macro expansions on statements/expressions with `stmt_expr_attributes` (in this case this error is correct).
I think we should pretend this is a "feature, not a bug" for now, land this PR and then possibly tweak feature gates so that https://github.com/rust-lang/rust/pull/49124 becomes gated only by `proc_macro`.

Reply:
I think I saw Serde relying on this, but I realize now it had to have been using `#[cfg]` which has been allowed and is probably being tested elsewhere already. So it's not an issue for now.